persist-cli/src/main/java/io/ballerina/persist/cmd/Generate.java

```diff
@@ -134,17 +146,69 @@ public void execute() {
             try {
                 BalProjectUtils.validateSchemaFile(file);
                 entityModule = BalProjectUtils.getEntities(file);
+                generatedSourceDirPath = Paths.get(this.sourcePath, BalSyntaxConstants.GENERATED_SOURCE_DIRECTORY);
+                HashMap<String, String> persistConfig = readPersistConfigurations(
+                        Paths.get(this.sourcePath, "Ballerina.toml"));
+                String submodule = "";
+                if (!persistConfig.get("module").equals(packageName)) {
+                    submodule = persistConfig.get("module").split("\\.")[1];
```

Comment:
We need extra validation here. what if we have `module name != package name` and not a prefix with the package name?
e.g:
package name = xyz
configured module name = yyy

Reply:
updated in [8ef0bc6](https://github.com/ballerina-platform/persist-tools/pull/148/commits/8ef0bc6316616b366e57d366857d0657fabd25e9)